### ISSUE TITLE
Duplicated content when using /routes when using CLI v4

### ISSUE THREAD

**Issue by charlier** [preactjs/preact-cli#1777]
This is referencing the [4.0.0-next.1](https://github.com/preactjs/preact-cli/issues/1579#issuecomment-1376339138) pre-release, so apologies if this is the wrong place to capture unexpected new features. Please shout if it would be happier on the request ticket!

**What is the current behaviour?**
If you have a component in /routes directory it seems to get weirdly duplicated, so you end up with 2 instances (notice two Get Started... sections in the screenshot below).
<img width="272" alt="Screenshot of duplicated content" src="https://user-images.githubusercontent.com/144476/214610237-df30ba24-77d6-48d7-bbff-2dd5ccae5396.png">

**Steps to Reproduce**
A very quick repo step is to just a do an `npm init preact-cli default defaultproj` & notice everything in the `<main>` section is there twice.
Or as a minimal code example,
```
import Home from './routes/home';

export default function App() {
    return (
	<main>
		<Home />
	</main>
    );
}
```

**What is the expected behaviour?**
Content not to be duplicated.

**Please mention any other relevant information**
Renaming the /routes directory (or moving the file), or downgrading to 3.4.5 as a workaround.

Tested on macOS 11.7.3, using NodeJS 16.17.1 (npm 8.15.0) & 18.13.0 (npm 8.19.3)

**Comment by rschristian** [comment]
This is a perfect place for v4 issues, thanks!

Are you seeing this when running the dev server or production builds?

**Comment by charlier** [comment]
This was with both production builds, and when running `preact watch`.
Having more of a poke around - I think the /routes directory might just be a symptom of the issue - and I'm in fact spinning up two instances of the app.

Looking at the source of the page (or opening up the generated index.html) I can see the bundle is included within the `<head>` element with a `defer` attribute, & again at the bottom of the page body with a `type="module"` attribute, so if I go ahead and remove the [latter](https://github.com/preactjs/preact-cli/blob/ffa026745d05fd0d0d9cc324561a7514270c5d14/packages/cli/src/resources/template.ejs#L22) from my template.ejs everything _just works_.

I'm guessing webpack thinks it's being helpful and adding one itself?

**Comment by rschristian** [comment]
Interesting, can reproduce with NPM but not Yarn... Weird dependency resolution going on I think, will need some time.

If you're not against using Yarn, maybe you could try that out?

**Comment by rschristian** [comment]
Finally tracked it down (sorry if you were getting updates, I ran into a few red herrings and was posting here in case I got called away) and it's a really weird issue.

Indeed, it's `html-webpack-plugin` trying to be helpful and injecting tags (scripts as you noticed, but also styles, meta, etc), but  what's weird is that this behavior seemingly differs between NPM and Yarn usage. With Yarn you get the old and expected behavior, but with NPM, extra scripts are found.

I don't really know how to solve this at the moment, will need to take it up with the plugin. We might be doing something wrong but this is baffling.
